src/Confluent.Kafka/Producer.cs

```diff
@@ -381,13 +380,16 @@ public long OutQueueLength
 
         /// <summary>
         ///     Wait until all outstanding produce requests, et.al, are completed. This should typically be done prior
-        //      to destroying a producer instance to make sure all queued and in-flight produce requests are completed
-        //      before terminating.
+        ///     to destroying a producer instance to make sure all queued and in-flight produce requests are completed
+        ///     before terminating. The wait is limited by the optional millisecondsTimeout parameter.
         /// </summary>
-        public void Flush(TimeSpan timeout)
```

Comment:
Yeah we did discuss. Though I think the optimal API is to be able to write `TimeSpan.FromMilliseconds(1000)` for clarity (i prefer this), even though it's longer than `1000`. `(int)TimeSpan.FromMilliseconds(1000).TotalMilliseconds` isn't very nice. Perf. isn't a consideration here.